### PR TITLE
cpufetch 1.04

### PR DESCRIPTION
**Pull request by p-linnane**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by p-linnane** [comment]
The 12-arm64 job on this PR keeps getting stuck in a queued state.

**Comment by p-linnane** [comment]
Looks like the test is consistently failing also on 12-arm64 only.

**Comment by SMillerDev** [comment]
`  [ERROR]: Found invalid cpu_family: 0x00000000`

**Comment by chenrui333** [comment]
unfortunately, I dont have a 12-arm to confirm if that is truly the issue.

**Comment by chenrui333** [comment]
logged an issue in the upstream for discussions.

**Comment by carlocab** [comment]
This is failing on 13-arm64 too (no failure since not bottled yet). I suspect this formula just doesn't quite work properly on our virtualised ARM runners. Even `brew` says the CPU is
```
CPU: quad-core 64-bit dunno
```
https://github.com/Homebrew/homebrew-core/actions/runs/4904192481/jobs/8757081355?pr=130340#step:2:110

File: Formula/cpufetch.rb [hunk 1]
```diff
@@ -39,15 +22,18 @@ def install
   end
 
   test do
-    actual = shell_output("#{bin}/cpufetch -d").each_line.first.strip
-
-    expected = if OS.linux?
-      "cpufetch v#{version} (Linux #{Hardware::CPU.arch} build)"
-    elsif Hardware::CPU.arm?
-      "cpufetch v#{version} (macOS ARM build)"
+    expected_result, line = if ENV["HOMEBREW_GITHUB_ACTIONS"].present? && Hardware::CPU.arm?
```
Comment (carlocab) [inline Formula/cpufetch.rb:25]:
```suggestion
    ephemeral_arm = ENV["HOMEBREW_GITHUB_ACTIONS"].present? &&
                    Hardware::CPU.arm?
                    MacOS.version > :big_sur
    expected_result, line = if ephemeral_arm
```

File: Formula/cpufetch.rb [hunk 2]
```diff
@@ -39,15 +22,18 @@ def install
...
+      [1, 1]
+    elsif OS.mac? && Hardware::CPU.intel?
+      [0, 1]
     else
-      "cpufetch is computing APIC IDs, please wait..."
+      [0, 0]
     end
+    actual = shell_output("#{bin}/cpufetch --debug", expected_result).lines[line].strip
```
Comment (carlocab) [inline Formula/cpufetch.rb:32]:
```suggestion
    actual = shell_output("#{bin}/cpufetch --debug 2>&1", expected_result).lines[line].strip
```

File: Formula/cpufetch.rb [hunk 3]
```diff
@@ -39,15 +22,21 @@ def install
...
 
   test do
-    actual = shell_output("#{bin}/cpufetch -d").each_line.first.strip
-
-    expected = if OS.linux?
-      "cpufetch v#{version} (Linux #{Hardware::CPU.arch} build)"
-    elsif Hardware::CPU.arm?
-      "cpufetch v#{version} (macOS ARM build)"
+    ephemeral_arm = ENV["HOMEBREW_GITHUB_ACTIONS"].present? &&
+                    Hardware::CPU.arm?
```
Comment (carlocab) [inline Formula/cpufetch.rb:26]:
```suggestion
                    Hardware::CPU.arm? &&
```

File: Formula/cpufetch.rb [hunk 4]
```diff
@@ -39,15 +22,21 @@ def install
...
-      "cpufetch is computing APIC IDs, please wait..."
+      [0, 0]
     end
+    actual = shell_output("#{bin}/cpufetch --debug 2>&1", expected_result).lines[line].strip
+
+    system_name = OS.mac? ? "macOS" : OS.kernel_name
+    arch = (OS.mac? && Hardware::CPU.arm?) ? "ARM" : Hardware::CPU.arch
+    expected = "cpufetch v#{version} (#{system_name} #{arch} build)"
 
     assert_equal expected, actual
```
Comment (carlocab) [inline Formula/cpufetch.rb:41]:
```suggestion
    assert_match expected, actual
```

**Pull request closed without merging**